File: backend/app/models/unit.py
from sqlalchemy import Boolean, Column, DateTime, Integer, String, func

from app.core.database import Base


class Unit(Base):
    """Don vi trong to chuc Lu doan (phong ban, tieu doan, dai doi, tram, BCH, cap uy).

    Admin quan ly duoc (them/sua/xoa). Moi User co the gan `unit_id` tro toi day.
    """

    __tablename__ = "units"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    name = Column(String(120), unique=True, nullable=False, index=True)
    # phong_ban | tieu_doan | dai_doi | tram | bch_lu_doan | cap_uy
    unit_kind = Column(String(30), nullable=False, server_default="phong_ban", index=True)
    description = Column(String(255), nullable=True)
    is_active = Column(Boolean, nullable=False, server_default="1")
    created_at = Column(DateTime, nullable=False, server_default=func.now())

File: frontend/src/types/unit.ts
export type UnitKind =
  | 'phong_ban'
  | 'tieu_doan'
  | 'dai_doi'
  | 'tram'
  | 'bch_lu_doan'
  | 'cap_uy'

export interface Unit {
  id: number
  name: string
  unit_kind: UnitKind
  description: string | null
  is_active: boolean
  created_at: string
  user_count: number
}

export interface UnitCreate {
  name: string
  unit_kind: UnitKind
  description?: string | null
  is_active: boolean
}

export const UNIT_KIND_LABELS: Record<UnitKind, string> = {
  phong_ban: 'Phòng / Ban',
  tieu_doan: 'Tiểu đoàn',
  dai_doi: 'Đại đội',
  tram: 'Trạm',
  bch_lu_doan: 'Ban chỉ huy Lữ đoàn',
  cap_uy: 'Cấp uỷ / Đảng bộ',
}
